src/components/InventoryTable/EntityTableToolbar.js

```diff
@@ -331,7 +331,10 @@ const EntityTableToolbar = ({
                         ...filter,
                         filterValues: {
                             placeholder: filter?.filterValues?.placeholder || `Filter by ${filter?.label?.toLowerCase()}`,
-                            ...filter?.filterValues,
+                            /* TODO: remove the OS check once RHEL Advisor has buildOSFilterConfig function call updated */
```

Comment:
@karelhala, @bastilian, I would like to know your opinion on this issue. The thing is, RHEL Advisor calls `buildOSFilterConfig` and overrides filter configurations including the OS filter. However, the function, in order to polyfill available OS filter values, used the deleted `operatingSystems` constant. This leads to the situation when Advisor generates the OS filter config with 0 available values.

The fix in this commit should detect this situation and apply the values from the API call (that Inventory made, not Advisor). This is a kludge, but we can revisit this scenario later on. This only happens to RHEL Advisor (other apps do not override config using `buildOSFilterConfig`.

Reply:
@gkarat I would go the other way around and align Advisor to the new API before going further with this PR. That way we don't have to keep any outstanding TODOs. However if that would have proven to be too complex we can go this way and align Advisor later (later meaning the sprint right after this gets merged)

Reply:
@karelhala, sounds reasonable. Let me try to take a look at the RHEL Advisor's code and see if we can easily rework this. I will link the PR to this one once I open.

Reply:
@karelhala, OK, this turned out to be possible. PTAL https://github.com/RedHatInsights/insights-advisor-frontend/pull/1027